docs/DeveloperGuide.md

```diff
@@ -1,34 +1,315 @@
+4. Download and open Intellij. You can download Intellij from [here](https://www.jetbrains.com/idea/)
+5. If you are not in the welcome screen, click `File` -> 'Close project'
+   to close the existing project dialog.
+6. Set up the correct JDK version for Gradle.
+    1. Click `Configure` -> `Project Defaults` -> `Project Structure`
+    2. Click `New...` and set it to the directory of the JDK.
+7. Click `Open or Import` in Intellij.
+8. Locate the `build.gradle` file, select it and click `OK`.
+9. If asked, choose to `Open as Project`.
+10. Click `OK` to accept the default settings.
+11. Right click on the `Duke.java` file and select `Run Duke.main()`.
+12. If the set up was done right, you should see this welcome message:
+    ````  
+     Welcome to 
+       _   _               _      _____                                _
+      | | | |   _ _       (_)    |_   _|     _ _    __ _      __      | |__     ___       _ _
+      | |_| |  | ' \      | |      | |      | '_|  / _` |    / _|     | / /    / -_)     | '_|
+       \___/   |_||_|    _|_|_    _|_|_    _|_|_   \__,_|    \__|_    |_\_\    \___|    _|_|_
+     _|"""""| _|"""""| _|"""""| _|"""""| _|"""""| _|"""""| _|"""""| _|"""""| _|"""""| _|"""""|
+     "`-0-0-' "`-0-0-' "`-0-0-' "`-0-0-' "`-0-0-' "`-0-0-' "`-0-0-' "`-0-0-' "`-0-0-' "`-0-0-'
+  
+    This is the list of your pinned tasks:
+  
+    Main Menu:
+    [1] Module Information
+    [2] CAP Simulator/Calculator
+    [3] Task Manager
+    [4] External Links
+    [5] Exit Program
+    ````
+---
+
+## 3. Design
+
+### 3.1 Architecture
+
+The `Duke` class contains the main method of our application.
+
+- At the start of the application, it initializes all the classes in sequence.
+
+We have 5 main classes for our key features.
+
+- `ModuleInfo`: contains methods for our module information features.
+- `HelpGraduation`: contains methods for our SU calculator/simulator.
+- `TaskManager`: contains methods for our task management features.
+- `Links`: contains methods for our links features.
+- `Storage`: Write data to, and loads data to hard disk
+
+![Diagram](diagrams/MainArchitecture.png)
+
+### 3.2 UI Class
+
+This class handles the interactions with the user and contains the methods for system output.
+
+---
+
+## 4. Implementation
+
+### 4.01 Add New Review
+
+This feature allows user to `add a review` for any of the modules they have taken.
+
+**NOTE**: This action will **overwrite** any existing reviews with the new review.
+
+E.g. Input:  
+`CS2113T is very fun:) I have learnt so many important concepts.`  
+`Overall 10/10 :)`  
+`/end`
+
+Output:
+
+```
+Woohoo~ Review added:  
+CS2113T is very fun:) I have learnt so many important concepts.  
+Overall 10/10 :) 
+
+Returning to module information menu...
+  ```
+Given below is the sequence diagram for the feature:  
+![Diagram](diagrams/addNewReview.png)  
```

Comment:
Maybe can consider breaking this sequence diagram into smaller fragments using ref frames